mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,415 @@
+* `num_neighbors : int` ➡ number of sampled neighbors
+* `edge_message_function_type: str` ➡ message function type, "identity" for concatenation or "mlp" for projection
+* `message_aggregator_type: str` ➡ message aggregator type, "mean" or "last"
+* `memory_updater_type: str` ➡ memory updater type, "gru" or "rnn"
+* `num_attention_heads: int` ➡ number of attention heads used if you define "graph_attn" as layer type
+* `learning_rate: float` ➡ learning rate for optimizer
+* `weight_decay: float` ➡ weight decay used in optimizer
+
+#### Usage:
+
+```cypher
+CALL tgn.set_params("self_supervised", 200, 2, "graph_attn", 100, 100, 20, 20, 100, 10, "identity", "last", "gru", 1) YIELD *;
+```
+
+
+### `update(edges)`
+
+#### Input:
+
+* `edges: mgp.List[mgp.Edges]` ➡ list of edges to preprocess, and if current batch size is big enough use for training or evaluation
+
+
+#### Usage:
+
+There are a few options here. 
+
+The best one is to create a **[trigger](https://memgraph.com/docs/memgraph/reference-guide/triggers)**, so every
+time an edge is added to graph, the trigger calls a procedure and makes an
+update.
+
+```cypher
+CREATE TRIGGER create_embeddings ON --> CREATE BEFORE COMMIT
+EXECUTE CALL tgn.update(createdEdges) RETURN 1;
+```
+
+The second option is to add all the edges and then call the algorithm with those
+edges:
+
+```cypher
+MATCH (n)-[e]->(m)
+WITH COLLECT(e) as edges
+CALL tgn.update(edges) RETURN 1;
+```
+
+
+### `get()`
+
+#### Output:
+
+* `node: mgp.Vertex` ➡ vertex in Memgraph database
+* `embedding: mgp.List[float]` ➡ low-dimensional representation of node in form of graph embedding
+
+#### Usage:
+
+```cypher
+CALL tgn.get() YIELD * RETURN *;
+```
+
+
+### `set_mode(mode)`
+
+With this function you can change mode of **temporal graph networks** to "eval" mode. Any new edges which arrive
+will **not** be used to **train** module, but to **evaluate** it's mean average precision on **evaluation** set.
+
+If you know when your stream will end, it is good option to change mode to "eval" and test how it performs. Once stream
+is done, good option will be to run training of few epochs to get best results.
+
+#### Input:
+
+* `mode: str` ➡ mode in which **temporal graph networks** should be working, "train" or "eval"
+
+
+#### Usage:
+
+```cypher
+CALL tgn.set_mode(eval) YIELD *;
+```
+
+
+### `get_results()`
+
+#### Output:
+
+* `epoch_num=mgp.Number` ➡ number of epoch in which "training" or "evaluation" was done
+* `batch_num=mgp.Number` ➡ number of batch inside of epoch in which "training" or "evaluation" was done
+* `batch_process_time=mgp.Number` ➡ time needed to process batch 
+* `accuracy=mgp.Number` ➡ mean average precision on current batch
+* `accuracy_type=str` ➡ type of MAP performed, "train" or "eval"
+
+
+#### Usage:
+
+```cypher
+CALL tgn.get_results() YIELD * RETURN *;
+```
+
+
+### `process_epochs(num_epochs)`
```

Comment:
This method is not explained well, please add description of method below this title